esphome/components/hbridge/light.py

```diff
@@ -0,0 +1,24 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import light, output
+from esphome.const import CONF_OUTPUT_ID, CONF_PIN_A, CONF_PIN_B
+
+hbridge_ns = cg.esphome_ns.namespace('hbridge')
+HBRIDGELightOutput = hbridge_ns.class_('HBRIDGELightOutput', cg.PollingComponent, light.LightOutput)
+
+CONFIG_SCHEMA = light.RGB_LIGHT_SCHEMA.extend({
+    cv.GenerateID(CONF_OUTPUT_ID): cv.declare_id(HBRIDGELightOutput),
+    cv.Required(CONF_PIN_A): cv.use_id(output.FloatOutput),
+    cv.Required(CONF_PIN_B): cv.use_id(output.FloatOutput),
+})
```

Comment:
Im not 100% sure about this, but you could try also extend `polling_component_schema` so users can override the `update_interval` in config to speed up and slow down the switching since you are already extending PollingComponent. You can set the default `1s` in the config and remove the constructor from the `.h` file

```suggestion
}).extend(cv.polling_component_schema('1s'))
```

Reply:
I tried polling_component_schema originally however it is not fast enough. I actually need the loop to run as fast as it can

Reply:
Ah I was getting confused about the PollingComponent and how you were using it.
So with this, both "sets" of lights would always be on?
It loops about 60 times a second because the ESPHome loop is 16ms intervals, so your `update` method will only be called once every 16ms even with the 1ms time on the `PollingComponent`

Reply:
Have you tried changing it to extend `Component` instead of `PollingComponent` and then override `loop` instead of `update`?

Reply:
Is this the recommended approach? 
I am happy with the refresh rate. The only problem is that it can flicker when it is running other background tasks like MQTT or WIFI things. However, I don’t want to impact these tasks so was happy with it.

Reply:
Yeah, there is no point in setting a PollingComponent unless it can be overridden by the user in yaml. And if you want it to switch between them as fast as possible, then using loop is better as there is more logic around the PollingComponent update times etc that has to do calculations. I guess you could tie into effects like string A on for 1 second then string B on for 1 second (looping) later on. But this currently would just appear as all lights being on right? Can you post a video of what it currently looks like?

Reply:
The PollingComponent is used for consistency. I was unable to stop flickering with the Component.
Here is a video of what it is and does - https://youtu.be/_sU-UKU6fV8